server/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -666,4 +659,8 @@ public static DataType<?> fromId(Integer id) {
             .findFirst()
             .orElse(DataTypes.UNDEFINED);
     }
+
+    public static boolean isArrayOfNulls(DataType<?> type) {
+        return type instanceof ArrayType<?> at && at.innerType() == UndefinedType.INSTANCE;
+    }
```

Comment:
```suggestion

    public static boolean isArrayOf(DataType<?> type, DataType<?> innerType) {
        return type instanceof ArrayType<?> at && at.innerType().id() == innerType.id();
    }
```

Not sure if worth it, but would make it a bit more generic.